### ISSUE TITLE
Trainig imagen-pytorch with one's own dataset + How many steps??

### ISSUE THREAD

**Issue by lucalevi** [lucidrains/imagen-pytorch#239]
Hi all 

I am trying to understand how this library works.
For the sake of simplicity, I am going to refer to the first code block in the repo immediately after "Usage"
https://github.com/lucidrains/imagen-pytorch#usage

As far as I've understood so far, the lines

`text_embeds = torch.randn(4, 256, 768).cuda()`
`images = torch.randn(4, 3, 256, 256).cuda() `

are mock lines, won't produce anything meaningful (noise images) and are to be substituted with one's own data.

The question is now: how?

I have downloaded (only) the first snappy.parquet file from 
https://huggingface.co/datasets/laion/laion400m/tree/main 
to keep thing simple for the moment

With the following code

`from fastparquet import ParquetFile`
`import snappy`
`def snappy_decompress(data, uncompressed_size):`
`    return snappy.decompress(data)`
`pf = ParquetFile('[very secret path]/part-00000-5b54c5d5-bbcf-484d-a2ce-0d6f73df1a36-c000.snappy.parquet') # filename includes .snappy.parquet extension`
`dff=pf.to_pandas()`

I created a pandas dataframe from the snappy.parquet file.
The dataframe has, most importantly, the columns URL and TEXT

Now, how do I pass these parameters to `text_embeds` and `images` from above?
Does one have to have actual images and not only URLs? Maybe using `img2dataset` first? 
https://github.com/rom1504/img2dataset 
(in that case this information should be added in the repo!)

How would the lines 
`text_embeds = ?`
`images = ?`
look like using a real dataset that works?

---

Finally, one more question about the iterations, here:
`for i in (1, 2):`
`    loss = imagen(images, text_embeds = text_embeds, unet_number = i)`
`    loss.backward()`

`# do the above for many many many many steps`

What does "do the above for many many many many steps" actually imply? 
What has one actually to do to do the above.... many many many many times?

All answers are appreciated!
And thanks a lot everybody and for sure @lucidrains for the great work you are all doing!
